python/ray/tests/test_actor_failures.py

```diff
@@ -1248,5 +1248,271 @@ def get_pid(self):
     assert ray.get(refs) == [3, 4, 5]
 
 
+def test_actor_cleanup_behavior(ray_start_regular):
+    """Test actor cleanup behavior: __ray_shutdown__ method is called."""
+    import atexit
+    import tempfile
+    import os
+
+    cleanup_file = tempfile.NamedTemporaryFile(delete=False)
+    cleanup_file.close()
+
+    atexit_file = tempfile.NamedTemporaryFile(delete=False)
+    atexit_file.close()
```

Comment:
we should use a fixture or `with` block for these to ensure they're deleted on test exit